Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift

```diff
@@ -0,0 +1,377 @@
+        paymentFlowClient.performThreeDSecureLookup(threeDSecureRequest) { lookupResult, error in
+            DispatchQueue.main.async {
+                guard let lookupResult, error == nil else {
+                    apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.failed")
+                    self.paymentFlowClientDelegate?.onPayment(with: nil, error: error)
+                    return
+                }
+
+                let threeDSecureVersion = lookupResult.lookup?.threeDSecureVersion ?? "2"
+                apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.3ds-version.\(threeDSecureVersion)")
+
+                self.threeDSecureRequestDelegate?.onLookupComplete(threeDSecureRequest, lookupResult: lookupResult) {
+                    let requiresUserAuthentication = lookupResult.lookup?.requiresUserAuthentication ?? false
+                    apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.challenge-presented.\(self.stringFor(requiresUserAuthentication))")
+                    self.process(lookupResult: lookupResult, configuration: configuration)
+                }
+            }
+        }
+    }
+    
+    private func performV2Authentication(with lookupResult: BTThreeDSecureResult) {
+        guard let apiClient = paymentFlowClientDelegate?.apiClient() else {
+            paymentFlowClientDelegate?.onPaymentComplete(nil, error: BTThreeDSecureError.invalidAPIClient)
+            return
+        }
+
+        threeDSecureV2Provider?.process(lookupResult: lookupResult) { result, error in
+            guard let result else {
+                apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.failed")
+                self.paymentFlowClientDelegate?.onPaymentComplete(nil, error: error)
+                return
+            }
+
+            self.logThreeDSecureCompletedAnalytics(forResult: lookupResult, apiClient: apiClient)
+            self.paymentFlowClientDelegate?.onPaymentComplete(result, error: error)
+        }
+    }
+    
+    private func logThreeDSecureCompletedAnalytics(forResult result: BTThreeDSecureResult, apiClient: BTAPIClient) {
+        let liabilityShiftPossible = result.tokenizedCard?.threeDSecureInfo.liabilityShiftPossible ?? false
+        apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.liability-shift-possible.\(stringFor(liabilityShiftPossible))")
+
+        let liabilityShifted = result.tokenizedCard?.threeDSecureInfo.liabilityShiftPossible ?? false
+        apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.liability-shifted.\(liabilityShifted)")
+
+        apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.completed")
+    }
+    
+    func stringFor(_ boolean: Bool) -> String {
+        boolean ? "true" : "false"
+    }
+}
+
+// MARK: - BTPaymentFlowRequestDelegate Protocol Conformance
+
+extension BTThreeDSecureRequest: BTPaymentFlowRequestDelegate {
+
+    /// :nodoc:
```

Comment:
What does this mean?

Reply:
We use Jazzy for our [reference docs](https://braintree.github.io/braintree_ios/current/). `:nodoc:` is Jazzy's way of excluding docs from the public reference documentation that is not for merchant consumption/use. This is public because the protocol is public to be used in both the Payment Flow and 3D Secure module.